src/platform/forms-system/src/js/validation.js

```diff
@@ -420,9 +420,11 @@ export function validateCurrentOrPastYear(errors, year) {
   }
 }
 
-export function validateMatch(field1, field2) {
+export function validateMatch(field1, field2, ignoreCase = false) {
   return (errors, formData) => {
-    if (formData[field1] !== formData[field2]) {
+    const flags = ignoreCase ? 'gi' : 'g';
+    const regexp = new RegExp(`^${formData[field2]}$`, flags);
+    if (!formData[field1]?.match(regexp)) {
```

Comment:
Instead of having a third argument that's a boolean, could we make it an object with a key so that it's usage is more explicit? For example:

```js
 export function validateMatch(field1, field2, { ignoreCase }) { 
```

```js

validateMatch('email', 'view:confirmEmail', { ignoreCase: true } )
```

And instead of a regex, what about using `.toLowerCase()`?

```js
  return (errors, formData) => {
    const transform = ignoreCase ? (val) => val.toLowerCcase() : (val)=>val;
    if (transform(formData[field1]) !== transform(formData[field2])) {
       errors[field2].addError('Please ensure your entries match'); 
     }
   };  
```

Reply:
For the record, I'm not pushing the specific `transform` idea, there's probably another way that doesn't use a no-op for most of the cases :upside_down_face:

Reply:
changed